core/src/test/scala/unit/kafka/server/RequestQuotaTest.scala

```diff
@@ -90,94 +102,139 @@ class RequestQuotaTest extends BaseRequestTest {
     }
   }
 
+  private def changeClientIdConfig(sanitizedClientId: String, configs: Properties): Unit = {
+    if (isKRaftTest()) {
+      val admin = createAdminClient()
+      admin.alterClientQuotas(Collections.singleton(
+        new ClientQuotaAlteration(
+          new ClientQuotaEntity(Map(ClientQuotaEntity.CLIENT_ID -> (if (sanitizedClientId == "<default>") null else sanitizedClientId)).asJava),
+          configs.asScala.map { case (key, value) => new ClientQuotaAlteration.Op(key, value.toDouble) }.toList.asJava)
+      )).all().get()
+    } else {
+      adminZkClient.changeClientIdConfig(sanitizedClientId, configs)
+    }
+  }
+
   @AfterEach
   override def tearDown(): Unit = {
     try executor.shutdownNow()
     finally super.tearDown()
   }
 
-  @Test
-  def testResponseThrottleTime(): Unit = {
-    for (apiKey <- RequestQuotaTest.ClientActions ++ RequestQuotaTest.ClusterActionsWithThrottle)
+  @ParameterizedTest(name = TestInfoUtils.TestWithParameterizedQuorumName)
+  @ValueSource(strings = Array("zk", "kraft"))
+  def testResponseThrottleTime(quorum: String): Unit = {
+    for (apiKey <- clientActions ++ clusterActionsWithThrottle)
       submitTest(apiKey, () => checkRequestThrottleTime(apiKey))
 
     waitAndCheckResults()
   }
 
-  @Test
-  def testResponseThrottleTimeWhenBothProduceAndRequestQuotasViolated(): Unit = {
+  @ParameterizedTest(name = TestInfoUtils.TestWithParameterizedQuorumName)
+  @ValueSource(strings = Array("zk", "kraft"))
+  def testResponseThrottleTimeWhenBothProduceAndRequestQuotasViolated(quorum: String): Unit = {
     submitTest(ApiKeys.PRODUCE, () => checkSmallQuotaProducerRequestThrottleTime())
     waitAndCheckResults()
   }
 
-  @Test
-  def testResponseThrottleTimeWhenBothFetchAndRequestQuotasViolated(): Unit = {
+  @ParameterizedTest(name = TestInfoUtils.TestWithParameterizedQuorumName)
+  @ValueSource(strings = Array("zk", "kraft"))
+  def testResponseThrottleTimeWhenBothFetchAndRequestQuotasViolated(quorum: String): Unit = {
     submitTest(ApiKeys.FETCH, () => checkSmallQuotaConsumerRequestThrottleTime())
     waitAndCheckResults()
   }
 
-  @Test
-  def testUnthrottledClient(): Unit = {
-    for (apiKey <- RequestQuotaTest.ClientActions) {
+  @ParameterizedTest(name = TestInfoUtils.TestWithParameterizedQuorumName)
+  @ValueSource(strings = Array("zk", "kraft"))
+  def testUnthrottledClient(quorum: String): Unit = {
+    for (apiKey <- clientActions) {
       submitTest(apiKey, () => checkUnthrottledClient(apiKey))
     }
 
     waitAndCheckResults()
   }
 
-  @Test
-  def testExemptRequestTime(): Unit = {
-    val actions = RequestQuotaTest.ClusterActions -- RequestQuotaTest.ClusterActionsWithThrottle -- RequestQuotaTest.Envelope
+  @ParameterizedTest(name = TestInfoUtils.TestWithParameterizedQuorumName)
+  @ValueSource(strings = Array("zk", "kraft"))
+  def testExemptRequestTime(quorum: String): Unit = {
+    // Exclude `DESCRIBE_QUORUM`, maybe it shouldn't be a cluster action
```

Comment:
https://issues.apache.org/jira/browse/KAFKA-15334 will fix this

Reply:
OK, I got it.